lib/src/legacy_nurouter.dart

```diff
@@ -212,7 +215,7 @@ abstract class NuRouter implements INuRouter {
     if (isFromNative) {
       _addNativePopCallBack(route);
     }
-    return route;
+    return route as Route<T>?;
```

Comment:
I have been noticed that type inference seems to be gotten worse, is this expected?

Reply:
In this case it can't infere Route<dynamic> as Route<T>?